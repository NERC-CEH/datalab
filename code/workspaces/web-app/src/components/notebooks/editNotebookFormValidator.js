// TODO - add unit tests
import validate from 'validate.js';

export const editConstraints = {
  displayName: {
    presence: true,
  },
  type: {
    presence: true,
  },
};

validate.formatters.reduxForm = errors => errors.reduce(errorReducer, {});

function errorReducer(accumulator, error) {
  accumulator[error.attribute] = error.error; // eslint-disable-line no-param-reassign
  return accumulator;
}

export const syncValidate = values => validate(values, editConstraints, { format: 'reduxForm' });

