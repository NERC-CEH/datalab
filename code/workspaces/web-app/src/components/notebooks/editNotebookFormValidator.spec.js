import { syncValidate } from './editNotebookFormValidator';

describe('editNotebookFormValidator', () => {
  describe('syncValidate', () => {
    it('should return empty errors object for valid form', () => {
      const values = {
        displayName: 'Display Name',
        type: 'jupyter',
      };

      expect(syncValidate(values)).toBeUndefined();
    });

    it('should return correct errors for empty form', () => {
      const values = {};
      expect(syncValidate(values)).toMatchSnapshot();
    });
  });
});
