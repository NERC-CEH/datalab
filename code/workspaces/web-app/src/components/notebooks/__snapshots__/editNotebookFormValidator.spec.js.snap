// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editNotebookFormValidator syncValidate should return correct errors for empty form 1`] = `
Object {
  "displayName": "Display name can't be blank",
  "type": "Type can't be blank",
}
`;
